engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -58,6 +60,9 @@ object JesBackend {
 
   def authGcsCredentialsPath(gcsPath: String): JesInput = JesLiteralInput(ExtraConfigParamName, gcsPath)
 
+  // Strictly speaking this could cause OOM issues but should be fine. Something to monitor
+  val GcsUploadThreadPool = ExecutionContext.fromExecutor(Executors.newCachedThreadPool())
```

Comment:
Curious to maybe chat IRL why this new thread pool is used, and not the actor system's thread pool. I've always been curious with items like the [`after`](http://doc.akka.io/docs/akka/current/scala/futures.html#After) pattern when one should use a separate thread pool.